framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4613,442 @@ private void makeGroundTargetType(
     addDefaultAnnotations(functionalType);
   }
 
+  /**
+   * Return true if {@code type} should be captured.
+   *
+   * <ul>
+   *   <li>{@code type} and {@code typeMirror} must both be declared types.
+   *   <li>{@code type} must not be a raw type, must not have an uninferred type argument, and must
+   *       have a wildcard as a type argument.
+   *   <li>{@code typeMirror} must have a captured type as a type argument.
+   * </ul>
+   *
+   * @param type annotated type to capture
+   * @param typeMirror underlying Java type
+   * @return true if {@code type} should be captured
+   */
+  private boolean shouldCapture(AnnotatedTypeMirror type, TypeMirror typeMirror) {
+    if (type.getKind() != TypeKind.DECLARED) {
+      return false;
+    }
+    if (typeMirror.getKind() != TypeKind.DECLARED) {
+      throw new BugInCF("Expected declared type mirror: %s, type: %s", typeMirror, type);
+    }
+
+    DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
+    AnnotatedDeclaredType uncapturedType = (AnnotatedDeclaredType) type;
+    if (capturedTypeMirror.getTypeArguments().isEmpty()) {
+      return false;
+    }
+
+    if (uncapturedType.wasRaw() || uncapturedType.containsUninferredTypeArguments()) {
```

Comment:
The past tense name `wasRaw` is in conflict with the present tense documentation "must not be a raw type".  (I guess it must not have been written by the programmer as a raw type, even if the type as provided to this method is not raw.  Is that correct?)

Reply:
Yes, but to clarify, `wasRaw` means that the underlying type is raw and the Checker Framework inferred type arguments.  The Checker Framework infers type arguments for all raw types.  `wasRaw` is probably a bad name.

Reply:
Could you please update the documentation of `wasRaw`?  Changing its name sounds like a good idea too -- that could be done in a different pull request and merged into this one.  Thanks!

Reply:
The name may be fine after all (now that I understand it), but updating the documentation would still be useful.  Thanks.